groovy-language-server/src/main/java/com/palantir/ls/groovy/GroovycWrapper.java

```diff
@@ -76,32 +82,44 @@
     private static final String JAVA_DEFAULT_OBJECT = "java.lang.Object";
 
     private final Path workspaceRoot;
-    private final CompilationUnit unit;
+    private final Path changedFilesRoot;
+    private final CompilerConfiguration config;
+
+    private CompilationUnit unit;
+
     // Maps from source file -> set of symbols in that file
     private Map<String, Set<SymbolInformation>> fileSymbols = Maps.newHashMap();
     // Maps from type -> set of references of this type
     private Map<String, Set<SymbolInformation>> typeReferences = Maps.newHashMap();
+    // Map from origin source filename to its changed version filename
+    private Map<Path, SourceWriter> originalSourceToChangedSource = Maps.newHashMap();
 
-    private GroovycWrapper(CompilationUnit unit, Path workspaceRoot) {
-        this.unit = unit;
+    private GroovycWrapper(Path workspaceRoot, Path changedFilesRoot, CompilerConfiguration config) {
         this.workspaceRoot = workspaceRoot;
+        this.changedFilesRoot = changedFilesRoot;
+        this.config = config;
+        unit = new CompilationUnit(config);
     }
 
     /**
      * Creates a new instance of GroovycWrapper.
+     *
      * @param targetDirectory the directory in which to put generated files
      * @param workspaceRoot the directory to compile
+     * @param changedFilesRoot the directory to stored temporary incrementally changed files
      * @return the newly created GroovycWrapper
      */
-    public static GroovycWrapper of(Path targetDirectory, Path workspaceRoot) {
-        Preconditions.checkNotNull(workspaceRoot, "workspaceRoot must not be null");
+    public static GroovycWrapper of(Path targetDirectory, Path workspaceRoot, Path changedFilesRoot) {
         Preconditions.checkNotNull(targetDirectory, "targetDirectory must not be null");
-        Preconditions.checkArgument(workspaceRoot.toFile().isDirectory(), "workspaceRoot must be a directory");
+        Preconditions.checkNotNull(workspaceRoot, "workspaceRoot must not be null");
+        Preconditions.checkNotNull(targetDirectory, "changedFilesRoot must not be null");
         Preconditions.checkArgument(targetDirectory.toFile().isDirectory(), "targetDirectory must be a directory");
```

Comment:
maybe do a file.exists() first to check whether the directory exists, or you can fix all the directory related things in #63